### DISCUSSION TITLE
Problems with 'git push origin'

### DISCUSSION THREAD

**Discussion by ghost** [tudelft/AE4353-Y24#12] [Q&A] [answered]
For anyone having the similar error: I had a problem with the 'git push origin' step. It's fixed now, because Yilun helped me, maybe he can explain how.  

The problem was the following error: 
![Error_gitpushorigin](https://github.com/user-attachments/assets/1741864c-f131-4a3b-9c5d-af3ba442111a)

**Comment by yilun-wu** [comment, marked as answer]
I think in the end, it was because we put the `ssh-agent` script under `.bash_profile` instead of `.bashrc` as instructed from the [steps](https://code.visualstudio.com/remote/advancedcontainers/sharing-git-credentials#_using-ssh-keys).

If you can do `git push origin` in WSL but not in the devcontainer, then make sure the [steps](https://code.visualstudio.com/remote/advancedcontainers/sharing-git-credentials#_using-ssh-keys) are correctly applied.

**Reply by Huizerd** [reply]
if `.bashrc` is the correct place, then the [steps](https://code.visualstudio.com/remote/advancedcontainers/sharing-git-credentials#_using-ssh-keys) are wrong, right? they suggest `.bash_profile`. also, I could only get the added lines to run without error if I comment out
```
eval `cat $HOME/.ssh/ssh-agent` > /dev/null
```

because it says the file `.ssh/ssh-agent` doesn't exist. do you also have that?
